sql/type.go

```diff
@@ -255,39 +255,21 @@ func (t numberT) Convert(v interface{}) (interface{}, error) {
 
 // Compare implements Type interface.
 func (t numberT) Compare(a interface{}, b interface{}) (int, error) {
-	if a == b {
+	ca, err := cast.ToInt64E(a)
```

Comment:
what if the number is unsigned? We're casting everything to int64, but numbers can be uint64

Reply:
@erizocosmico done